File: Homework evaluation/JAVASCRIPT EXPRESSJS OVERVIEW - EXERCISES/3/Homework/Calculator Skeleton/models/Calculator.js
function Calculator(leftOperand, operator, rightOperand) {
    this.leftOperand = leftOperand;
    this.operator = operator;
    this.rightOperand = rightOperand;

    this.calculateResult = function () {
        let result = 0;

        switch (this.operator){
            case "+": result = this.leftOperand + this.rightOperand;
                break;
            case "-": result = this.leftOperand - this.rightOperand;
                break;
            case "*": result = this.leftOperand * this.rightOperand;
                break;
            case "/": result = this.leftOperand / this.rightOperand;
                break;
            case "^": result = Math.pow(this.leftOperand, this.rightOperand);
                break;
            case "Rnd": result = this.leftOperand * this.rightOperand / 12 * 27 + 1 * 2007 / 3;
                break;
        }
        return result;
    }
}
module.exports = Calculator;

File: Homework evaluation/JAVA BASIC WEB - EXERCISES - NO PHYSICAL PRESENCE/Homework 4/Homework 4/src/main/java/com/softuni/entity/Calculator.java
package com.softuni.entity;

public class Calculator {
    private double leftOperand;
    private double rightOpererand;
    private String operator;

    public Calculator(double leftOperand, double rightOpererand, String operator) {
        this.leftOperand = leftOperand;
        this.rightOpererand = rightOpererand;
        this.operator = operator;
    }

    public double getLeftOperand() {
        return leftOperand;
    }

    public void setLeftOperand(double leftOperand) {
        this.leftOperand = leftOperand;
    }

    public double getRightOpererand() {
        return rightOpererand;
    }

    public void setRightOpererand(double rightOpererand) {
        this.rightOpererand = rightOpererand;
    }

    public String getOperator() {
        return operator;
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public double calculateResult(){
        double result = 0.0;
        double leftOperand = this.leftOperand;
        double rightOperand = this.rightOpererand;
        String operator = this.operator;
        switch(operator){
            case "+":
                result = leftOperand + rightOperand;
                break;
            case "-":
                result = leftOperand - rightOperand;
                break;
            case "*":
                result = leftOperand * rightOperand;
                break;
            case "/":
                result = leftOperand / rightOperand;
                break;
            case "anomalify":
                result = (leftOperand + rightOperand) * 69 / 420;
                break;
        }
        return result;
    }
}
